python/artm/artm_model.py

```diff
@@ -1011,6 +1023,185 @@ def __repr__(self):
+        params['num_document_passes'] = self._num_document_passes
+        params['reuse_theta'] = self._reuse_theta
+        params['theta_columns_naming'] = self._theta_columns_naming
+        params['seed'] = self._seed
+        params['show_progress_bars'] = self._show_progress_bars
+        params['topic_names'] = self._topic_names
+        params['class_ids'] = self._class_ids
+        params['model_pwt'] = self._model_pwt
+        params['model_nwt'] = self._model_nwt
+        params['theta_name'] = self._theta_name
+        params['synchronizations_processed'] = self._synchronizations_processed
+        params['num_online_processed_batches'] = self._num_online_processed_batches
+        params['initialized'] = self._initialized
+
+        regularizers = {}
+        for name, regularizer in iteritems(self._regularizers.data):
+            tau = None
+            gamma = None
+            try:
+                tau = score.tau
+                gamma = score.gamma
+            except:
+                pass
+            regularizers[name] = [str(regularizer.config), tau, gamma]
+        params['regularizers'] = regularizers
+
+        scores = {}
+        for name, score in iteritems(self._scores.data):
+            model_name = None
+            try:
+                model_name = score.model_name
+            except:
+                pass
+            scores[name] = [str(score.config), model_name]
+
+        params['scores'] = scores
+
+        with open(os.path.join(data_path, PARAMETERS_FILENAME_JSON), 'w') as fout:
+            json.dump(params, fout)
+
+        # save parameters in binary format
+        regularizers = {}
+        for name, regularizer in iteritems(self._regularizers._data):
+            regularizers[name] = [regularizer._config_message.__name__,
+                                  regularizer.config.SerializeToString()]
+
+            tau = None
+            gamma = None
+            try:
+                tau = regularizer.tau
+                gamma = regularizer.gamma
+            except:
+                pass
+
+            if tau is not None:
+                regularizers[name].append(tau)
+                if gamma is not None:
+                    regularizers[name].append(gamma)
+
+        params['regularizers'] = regularizers
+
+        scores = {}
+        for name, score in iteritems(self._scores._data):
+            scores[name] = [score._config_message.__name__,
+                            score.config.SerializeToString()]
+
+            model_name = None
+            try:
+                model_name = score.model_name
+            except:
+                pass
+            if model_name is not None:
+                scores[name].append(model_name)
+
+        params['scores'] = scores
+
+        with open(os.path.join(data_path, PARAMETERS_FILENAME_BIN), 'wb') as fout:
+            pickle.dump(params, fout)
+
 
 def version():
     return ARTM(num_topics=1).library_version
+
+
+def load_artm_model(data_path):
```

Comment:
OK, it seems reasonable. By the way, there are plenty of choices:

* The aforementioned `read_csv` method can return not only `DataFrame`, but also `TextParser` (i.e. some kind of iterator);
* `numpy` library uses functions, not methods, for both loading and saving, and in both cases it operates on array;
* `scipy` uses the same policy for sparse matrices as `numpy`; in their case, there are more than one possible type of argument;